tests/Models/User.php

```diff
@@ -109,6 +109,14 @@ protected function username(): Attribute
         );
     }
 
+    protected function skills(): Attribute
+    {
+        return Attribute::make(
+            get: fn ($value) => $value,
+            set: fn ($values) => array_unique($values)
```

Comment:
If you set the field name as key, the patch is not needed.

```suggestion
            set: fn ($values) => ['skills' => array_unique($values)]
```

Reply:
Yes I know, but it's not the best solution and it can lead to a bug when the developer renames the `skills` field. I prefer to fix this issue in the right way. Since this package is made on top of eloquent, it would be better to avoid making unexpected behavior.

Reply:
For the risk in refactoring, the method could be:
```suggestion
            set: fn ($values) => [__FUNCTION__ => array_unique($values)]
```

I'll have a new look at the solution.

Reply:
I want to fix this issue without determining a key because I believe it's not the correct way. I know I can set the key dynamically but it's not what I am trying to.

Reply:
Unfortunately, Laravel designed the functionality this way, without considering that a field can contain an array. Considering on the one hand that a field name can be an int (see #2642), and on the other that the array you want to store can be anything (a sequential list, disordered int or string keys), you can't detect properly without introducing random behavior depending on the value.

I understand that this is not an ideal experience for the developer. But why should `[1, 2, 3]` be treated as a single value and `['foo' => 1, 2, 9 => 3]` as a definition of the root fields `foo`, `0` and `9`?